### PR TITLE
Fix MERGE ROWID_VAR handling for PG17

### PR DESCRIPTION
**Pull request by svenklemm**
Disable-check: force-changelog-file

### CODE REVIEW

File: test/expected/ts_merge-17.out [hunk 1]
```diff
@@ -2549,3 +2539,16 @@ DROP USER regress_merge_privs;
...
 \o
 :DIFF_CMD
+--- Base pg table results
++++ Hyperatable results
+@@ -326,7 +326,9 @@
+ 
+  delta | sid | balance | tid | balance 
+ -------+-----+---------+-----+---------
+-(0 rows)
++   -10 |  -1 |      -1 |  -1 |     -11
```
Comment (fabriziomello) [inline test/expected/ts_merge-17.out:2549]: I think this test result is unexpected, no?

Comment (svenklemm) [reply test/expected/ts_merge-17.out:2549]: ts_merge test is huge mess, lots of errors and the non-ht vs ht dont even work with equivalent datasets

**Review by antekresic — APPROVED** [review]
lgtm

File: test/expected/ts_merge-17.out (hunk 1 above)
Comment (svenklemm) [reply test/expected/ts_merge-17.out:2549]: ts_merge needs another look, there is something wrong with that test that is not pg17 related

Comment (svenklemm) [reply test/expected/ts_merge-17.out:2549]: will be done in followup-pr

**Review by fabriziomello — APPROVED** [review]

**Pull request merged**